src/base/bittorrent/infohash.h

```diff
@@ -30,39 +30,101 @@
 
 #include <libtorrent/sha1_hash.hpp>
 
+#include <QByteArray>
+#include <QHash>
 #include <QMetaType>
 #include <QString>
 
 namespace BitTorrent
 {
+    template <std::ptrdiff_t N>
```

Comment:
why not use `int`? `ptrdiff_t` is for pointer arithmetic isn't?

Reply:
I would prefer using `int` in our code base if it isn't too much trouble.

Reply:
>I would prefer using `int` in our code base if it isn't too much trouble.

Ok, done.